service/history/replication/stream.go

```diff
@@ -56,3 +59,21 @@ func ClusterIDToClusterNameShardCount(
 	}
 	return "", 0, serviceerror.NewInternal(fmt.Sprintf("unknown cluster ID: %v", clusterID))
 }
+
+func GetErrorHandledAndRetriedEventLoop(originalEventLoop func() error, streamStopper func(), logger log.Logger, metricsHandler metrics.Handler) func() {
+	return func() {
```

Comment:
do we need to return a func() here? normal function won't work?

Reply:
It doesn't need to be a func. Let me know if you have strong opinion on this.

Reply:
then lets not do it. kind of weird.